Mathlib/Data/Fin/Tuple/Sort.lean

```diff
@@ -114,14 +115,67 @@ theorem sort_lt_at_start_of_monotone {α} [LinearOrder α] (m : ℕ) (f : Fin m
     (j : Fin m) (h : j < Fintype.card {i // f i ≤ a}) :
     f j ≤ a := by
   contrapose! h
-  have := Fintype.card_subtype_compl (¬f · ≤ a)
-  simp_rw [not_not, not_le] at this
-  rw [this, Fintype.card_fin, tsub_le_iff_tsub_le, Fintype.card_subtype]
-  refine le_trans (by simp) (Finset.card_mono $ show Finset.Ici j ≤ _ from fun k hk ↦ ?_)
-  simp only [Finset.mem_Ici] at hk
-  exact Finset.mem_filter.mpr ⟨Finset.mem_univ _, h.trans_le (h_sorted hk)⟩
+  rw [← Fin.card_Iio, Fintype.card_subtype]
+  refine Finset.card_mono (fun i => Function.mtr ?_)
+  simp_rw [Finset.mem_filter, Finset.mem_univ, true_and, Finset.mem_Iio]
+  exact fun hij => (h.trans_le <| h_sorted <| le_of_not_lt hij).not_le
 /- Proofs by Ruben Van de Velde, Eric {Rodriguez and Wieser} -/
 
+def Equiv.subtypeSubtype_comm  {α : Type} (p q : α → Prop) :
+    (Subtype fun x => (p x) ∧ (q x)) ≃ (Subtype fun x => (q x) ∧ (p x)) := by
+  simp_rw [and_comm]
+  apply Equiv.refl
```

Comment:
Using `rw` in `Type` is generally a bad idea.

There should be a `subtypeCongr` or similar which makes this a trivial consequence.